nutype_macros/src/common/gen/error.rs

```diff
@@ -11,12 +11,19 @@ pub fn gen_error_type_name(type_name: &TypeName) -> ErrorTypeName {
     ErrorTypeName::new(ident)
 }
 
-// NOTE: There is no `::core::error::Error` yet in stable Rust.
-// So for `no_std` we just don't implement `Error` trait.
+// NOTE: `::core::error::Error` is stable only for rust >= 1.81.0.
 #[allow(unused_variables)]
 pub fn gen_impl_error_trait(error_type_name: &ErrorTypeName) -> TokenStream {
     cfg_if! {
-        if #[cfg(feature = "std")] {
+        if #[cfg(ERROR_IN_CORE)] {
```

Comment:
Since the implementation is the same we should have something like 
`#[cfg(ERROR_IN_CORE)] || #[cfg(feature = "std")] ` instead of another `else if` branch that generates the same code.

If `#[cfg(ERROR_IN_CORE)] || #[cfg(feature = "std")] ` by some reason is not possible, then we should still DRY it up, by extracting the generated code in a var.

Reply:
Its not exactly the same implementation due to `std` and `core` imports.
I tried to DRY it up but couldn't at the time (partly because I didn't even think of `#[cfg(ERROR_IN_CORE)] || #[cfg(feature = "std")]`).

I saw a bunch of similar PRs about `std` vs `core` in the mean time, I'll give it another go 👍.

Reply:
well 😓 
<img width="1448" alt="image" src="https://github.com/user-attachments/assets/e724caec-09bf-4f71-a09a-f1b5aeee35c2">
<img width="1448" alt="image" src="https://github.com/user-attachments/assets/1a6b62f1-0f08-49bd-9952-9e8eef39a986">
<img width="1448" alt="image" src="https://github.com/user-attachments/assets/76121b78-9992-4e48-9b95-57eff1504658">

And I can't find any infos on how to do an OR operation, reading the source code I don't see any option to do it (but I could've missed it, I'm not really good with macros 😂).
The `not()` operation doesn't seem to be possible either.

I have another ideal, I'll keep you updated !